File: easy/data_recovery.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp $line;
    my ( $a, $b ) = split( /;/, $line );
    my @l = split( / /, $a );
    my @r = split( / /, $b );
    for ( my $i = 1 ; $i <= scalar @l ; $i++ ) {
        push( @r, $i ) if ( !grep { $_ == $i } @r );
    }
    my @s;
    for ( my $i = 0 ; $i < scalar @l ; $i++ ) {
        $s[ @r[$i] - 1 ] = @l[$i];
    }
    printf( "%s\n", join( ' ', @s ) );
}
close(INFILE);

File: easy/data_recovery.rb
File.open(ARGV[0]).each_line do |line|
  next unless line.include?(';')
  s = line.split(';')
  s0 = s[0].split
  s1 = s[1].split.map(&:to_i)
  (1..s0.length).each do |i|
    s1 << i unless s1.include?(i)
  end
  puts s0.sort_by.with_index { |_, i| s1[i] }.join(' ')
end
